tensorflow/java/src/main/java/org/tensorflow/op/OperationHelper.java

```diff
@@ -0,0 +1,147 @@
+import org.tensorflow.OperationBuilder;
+import org.tensorflow.Output;
+
+/**
+ * A helper for wrapping a new operation.
+ *
+ * <p>It provides utilities for wrappers to build a single operation and collect its output(s)
+ * afterward.
+ *
+ * <pre>{@code
+ * public final class Unique extends AbstractSingleOp {
+ *    ...
+ *    public static Unique create(Scope s, Input x) {
+ *        OperationHelper unique = OperationHelper.create(s, "Unique");
+ *        unique.builder().input(x.asOutput());
+ *        return new Unique(unique);
+ *    }
+ *    ...
+ *    private Unique(OperationHelper unique) {
+ *        super(unique.operation());
+ *        y = unique.nextOutput();
+ *        idx = unique.nextOutput();
+ *    }
+ * }
+ * }</pre>
+ *
+ * <p>{@code OperationHelper} objects are designed for a single usage and should be discarded once
+ * the operation wrapper has been created. They are <b>not</b> thread-safe.
+ */
+public final class OperationHelper {
+
+  /**
+   * Creates a helper for a new operation.
+   *
+   * <p>By default, the name of the operation is its {@code opType} in lowercase, unless a different
+   * name has been provided using {@link Scope#withName(String)}.
+   *
+   * @param scope operation scope
+   * @param opType operation type name
+   * @return new instance of this class
+   */
+  public static OperationHelper create(Scope scope, String opType) {
+    return new OperationHelper(
+        scope.graph().opBuilder(opType, scope.makeOpName(opType.toLowerCase())));
+  }
+
+  /**
+   * Returns the builder for the operation.
+   *
+   * <p>Wrappers can add inputs and attributes to the builder to configure the underlying operation.
+   * After the operation has been built (i.e. the first time {@link #operation()} is called), any
+   * further modifications to the builder are ignored.
+   *
+   * <p>Additionally, the {@code build()} method of the builder must <b>not</b> be called explicitly
+   * outside this class, {@link #operation()} must be used method instead.
+   *
+   * @return operation builder
+   */
+  public OperationBuilder builder() {
+    return builder;
+  }
+
+  /**
+   * Retrieve and optionally build the operation.
+   *
+   * <p>If the operation is retrieved for the first time, it will be built with inputs and
+   * attributes collected so far by the operation {@link #builder()}.
+   *
+   * @return an operation
+   * @see {@link #builder()}.
+   */
+  public Operation operation() {
+    if (unsafeOperation == null) {
+      unsafeOperation = builder.build();
+    }
+    return unsafeOperation;
+  }
+
+  /**
+   * Returns the next output of the operation.
+   *
+   * <p>All outputs collected with this method or with {@link #nextOutputList(String)} are retrieved
+   * sequentially, like an iterator would do.
+   *
+   * <p>This method invokes {@link #operation()} internally to build the operation if it has not
+   * been done yet.
+   *
+   * @return an output
+   */
+  public Output nextOutput() {
```

Comment:
If we do keep `OperationHelper` (see above, I'm tempted to avoid it altogether) - then I'd suggest against a `nextOutput`. I think it will be clearer and more explicit to not rely on the ordering of calls and do something like:

```
y = unique.operation().output(0);
idx = unique.operation().output(1);
```